validate.js

```diff
@@ -0,0 +1,126 @@
+  }
+}
+
+function removeErrorMessage(errorClass, element) {
+  let errorMessage = document.querySelector(`.${errorClass}`);
+  if (errorMessage) {
+    errorMessage.remove();
+    element.classList.remove('errorOutline');
+  }
+}
+
+// validation eventHandler
+function validateEmail() {
+  const regex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+  removeErrorMessage('emailErrorEmpty', email);
+  removeErrorMessage('emailErrorForm', email);
+  if (email.value.trim() === "") {
+    showErrorMessage(email, "이메일을 입력해주세요.", 'emailErrorEmpty');
+  } else if (!regex.test(email.value)) {
+    showErrorMessage(email, "잘못된 이메일 형식입니다.", 'emailErrorForm');
+  }
+  activateButton();
+}
+
+function validateName() {
+  removeErrorMessage('nameErrorEmpty', nickname);
+  if (nickname.value.trim() === "") {
+    showErrorMessage(nickname, "닉네임을 입력해주세요.", 'nameErrorEmpty');
+  }
+  activateButton();
+}
+
+function validatePw() {
+  removeErrorMessage('pwErrorEmpty', password);
+  removeErrorMessage('pwErrorShort', password);
+  if (password.value.trim() === "") {
+    showErrorMessage(password, "비밀번호를 입력해주세요.", 'pwErrorEmpty');
+  } else if (password.value.length < 8) {
+    showErrorMessage(password, "비밀번호를 8자 이상 입력해주세요.", 'pwErrorShort');
+  }
+  activateButton();
+  validatePwConfirm();
+}
+
+function validatePwConfirm() {
+  if (!passwordConfirm) return;
+  removeErrorMessage('pwConfirmErrorEmpty', passwordConfirm);
+  removeErrorMessage('pwConfirmError', passwordConfirm);
+  if (passwordConfirm.value.trim() === "") {
+    showErrorMessage(passwordConfirm, "비밀번호를 입력해주세요.", 'pwConfirmErrorEmpty');
+  } else if (password.value.trim() !== passwordConfirm.value.trim()) {
+    showErrorMessage(passwordConfirm, "비밀번호가 일치하지 않습니다.", 'pwConfirmError');
+  }
+  activateButton();
+}
+
+// button activation eventHandler
+function activateButton() {
+  const errors = document.querySelectorAll('.errorColor');
+  if (errors.length === 0 && 
+      email.value !== "" &&
+      password.value !== "" &&
+      (submitBtn.textContent === '로그인' || 
+        (nickname.value !== "" && 
+          passwordConfirm && passwordConfirm.value !== ""))) {
+    submitBtn.classList.remove('inactive-btn');
+    submitBtn.classList.add('active-btn');
+    if (submitBtn.textContent === '로그인') {
+      submitBtn.href = "/pages/items/items.html";
+    } else {
+      submitBtn.href = "/pages/signup/signup.html";
+    }
+  } else {
+    submitBtn.classList.remove('active-btn');
+    submitBtn.classList.add('inactive-btn');
+    submitBtn.removeAttribute('href');
+  }
+}
+
+// password Visibility eventHandler
+function togglePwVisibility(target, inputElement) {
+  if (inputElement.type === 'password') {
+    inputElement.type = 'text';
+    target.src = "/assets/icons/btn_visibility.png";
+  } else {
+    inputElement.type = 'password';
+    target.src = "/assets/icons/btn_visibility_off.png";
+  }
+}
+
+// login eventListener
+email.addEventListener('blur', validateEmail);
```

Comment:
같은 맥락의 코드는 서로 가깝게 위치시켜주면 가독성을 높여줍니다. 

```suggestion
function validateName() {
  removeErrorMessage('nameErrorEmpty', nickname);
  if (nickname.value.trim() === "") {
    showErrorMessage(nickname, "닉네임을 입력해주세요.", 'nameErrorEmpty');
  }
  activateButton();
}
const email = document.querySelector('#email');
email.addEventListener('blur', validateEmail);
```

Reply:
비슷한 역할을 하는 `focusout` 이벤트가 있습니다. 어떤 차이가 있을까요?
( https://mygumi.tistory.com/321 )

캡쳐링/버블링을 이용해 좀 더 좋은 코드를 작성 할 수 있으므로 `focusout`을 사용하는게 더 모던한 것 같습니다.